docs2/site/docs/migrations/migration5.md

```diff
@@ -570,6 +570,36 @@ var options = new ExecutionOptions
 var result = await executer.ExecuteAsync(options);
 ```
 
+### 19. Subscription support improved
+
+Support for subscriptions has been moved from the `GraphQL.SystemReactive` nuget package directly into
+the main `GraphQL` package. There is no need to use `SubscriptionDocumentExecuter` (removed), and the default
+document executer will support subscriptions without overriding `SelectExecutionStrategy`.
+
+The new implementation of `SubscriptionExecutionStrategy` supports some new features and bug fixes:
+
+1. Serial execution of data events' field resolvers is supported by passing an instance of
+   `SerialExecutionStrategy` to the constructor. As before, parallel execution is default.
+
+2. Errors and output extensions are returned along with data events.
+
+3. Memory leaks have been eliminated in the case of errors, output extensions, metrics being enabled,
```

Comment:
Where was the leak?

Reply:
As long as the subscription was active, metrics would continually be added to the execution context, for instance.  It would clear eventually, when the subscription was disposed.  For server-to-server subscriptions, where it is not expected that the connection should close at all, this could (eventually) crash the server.

Reply:
Similarly, errors and output extensions would continually add to the context and never be cleared until the subscription ended.

Reply:
Ah, I think I get it. Subscription handles data event in the same way as document executer - it executes node tree and calls resolvers (data object acts as root source) walking up down that tree. So each resolver _may_ enrich metrics (and other writable stuff) and so this shared (via closure) `context` object will grow more and more. Did I understand it right?

Reply:
Yes

Reply:
And metrics middleware would continually add shared data for every single field's execution, for every data event, which could easily use a lot of resources, depending on how complex the result was and how long the subscription was active.